apps/vs-code-designer/src/app/commands/createNewProject/createProjectSteps/ZipFileStep.ts

```diff
@@ -0,0 +1,68 @@
+import { unzipLogicAppArtifacts } from '../../../utils/taskUtils';
+import * as vscode from 'vscode';
+import * as path from 'path';
+import * as os from 'os';
+import * as fs from 'fs';
+import { localize } from '../../../../localize';
+import { AzureWizardPromptStep } from '@microsoft/vscode-azext-utils';
+import type { IAzureQuickPickItem } from '@microsoft/vscode-azext-utils';
+import type { IFunctionWizardContext } from '@microsoft/vscode-extension-logic-apps';
+
+import * as rimraf from 'rimraf';
+
+export class ZipFileStep extends AzureWizardPromptStep<IFunctionWizardContext> {
+  private zipContent: Buffer | Buffer[];
+  private targetDirectory: string;
+  public static zipFilePath: string;
+  private wizardContext: IFunctionWizardContext;
+
+  public hideStepCount = true;
+  public supportsDuplicateSteps = false;
+
+  // will likely need to change the location of the start zip path
+  public static async setZipFilePath(_context: Partial<IFunctionWizardContext>): Promise<void> {
+    const fileUris = await vscode.window.showOpenDialog({
+      canSelectMany: false,
+      defaultUri: vscode.Uri.file(path.join(os.homedir(), 'Downloads')),
+      openLabel: localize('selectZipFile', 'Select a zip file'),
+    });
+
+    if (fileUris && fileUris.length > 0) {
+      ZipFileStep.zipFilePath = fileUris[0].fsPath;
+    }
+  }
+
+  public async prompt(context: IFunctionWizardContext): Promise<void> {
+    this.wizardContext = context;
+    await ZipFileStep.setZipFilePath(context);
+    await this.getZipFiles();
+  }
+
+  public shouldPrompt(_context: IFunctionWizardContext): boolean {
+    return this.zipContent === undefined;
+  }
+
+  public async getZipFiles(): Promise<IAzureQuickPickItem<Buffer>[]> {
+    if (!this.wizardContext) {
+      console.error('wizardContext is not set in getzipfILes.');
```

Comment:
update the casing.

Reply:
Jaden wrote this part, I will remove it as I haven't reached a point where the wizard context is not set.